File: lib_common/src/main/java/com/yyxnb/common/SPUtils.java
package com.yyxnb.common;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * 保存工具类
 *
 * @author yyx
 */
public class SPUtils {

    /**
     * 保存在手机里面的文件名
     */
    private static final String FILE_NAME = "share_date";
    private static final Context CONTEXT = AppConfig.getInstance().getContext();

    /**
     * 保存数据的方法，我们需要拿到保存数据的具体类型，然后根据类型调用不同的保存方法
     *
     * @param key
     * @param object
     */
    public static void setParam(String key, Object object) {

        String type = object.getClass().getSimpleName();
        SharedPreferences sp = CONTEXT.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sp.edit();

        switch (type) {
            case "String":
                editor.putString(key, (String) object);
                break;
            case "Integer":
                editor.putInt(key, (Integer) object);
                break;
            case "Boolean":
                editor.putBoolean(key, (Boolean) object);
                break;
            case "Float":
                editor.putFloat(key, (Float) object);
                break;
            case "Long":
                editor.putLong(key, (Long) object);
                break;
            default:
                break;
        }
        editor.apply();
    }

    /**
     * 得到保存数据的方法，我们根据默认值得到保存的数据的具体类型，然后调用相对于的方法获取值
     *
     * @param key
     * @param defaultObject
     * @return
     */
    public static Object getParam(String key, Object defaultObject) {
        String type = defaultObject.getClass().getSimpleName();
        SharedPreferences sp = CONTEXT.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);

        switch (type) {
            case "String":
                return sp.getString(key, (String) defaultObject);
            case "Integer":
                return sp.getInt(key, (Integer) defaultObject);
            case "Boolean":
                return sp.getBoolean(key, (Boolean) defaultObject);
            case "Float":
                return sp.getFloat(key, (Float) defaultObject);
            case "Long":
                return sp.getLong(key, (Long) defaultObject);
            default:
                break;
        }
        return null;
    }

    /**
     * 清除所有数据
     */
    public static void clearAll() {
        SharedPreferences sp = CONTEXT.getSharedPreferences(FILE_NAME,
                Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sp.edit();
        editor.clear().apply();
    }

    /**
     * 清除指定数据
     */
    public static void clear(String key) {
        SharedPreferences sp = CONTEXT.getSharedPreferences(FILE_NAME,
                Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sp.edit();
        editor.remove(key);
        editor.apply();
    }
}

File: lib_common/src/main/java/com/yyxnb/common/utils/SPUtils.kt
package com.yyxnb.common.utils

import android.content.Context
import com.yyxnb.common.CommonManager.getContext

/**
 * 保存工具类
 *
 * @author yyx
 */
object SPUtils {
    /**
     * 保存在手机里面的文件名
     */
    private const val FILE_NAME = "share_date"
    private val CONTEXT = getContext()

    /**
     * 保存数据的方法，我们需要拿到保存数据的具体类型，然后根据类型调用不同的保存方法
     *
     * @param key
     * @param object
     */
    fun setParam(key: String?, `object`: Any) {
        val type = `object`.javaClass.simpleName
        val sp = CONTEXT.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
        val editor = sp.edit()
        when (type) {
            "String" -> editor.putString(key, `object` as String)
            "Integer" -> editor.putInt(key, (`object` as Int))
            "Boolean" -> editor.putBoolean(key, (`object` as Boolean))
            "Float" -> editor.putFloat(key, (`object` as Float))
            "Long" -> editor.putLong(key, (`object` as Long))
            else -> {
            }
        }
        editor.apply()
    }

    /**
     * 得到保存数据的方法，我们根据默认值得到保存的数据的具体类型，然后调用相对于的方法获取值
     *
     * @param key
     * @param defaultObject
     * @return
     */
    fun getParam(key: String, defaultObject: Any): Any? {
        val type = defaultObject.javaClass.simpleName
        val sp = CONTEXT.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
        when (type) {
            "String" -> return sp.getString(key, defaultObject as String)
            "Integer" -> return sp.getInt(key, (defaultObject as Int))
            "Boolean" -> return sp.getBoolean(key, (defaultObject as Boolean))
            "Float" -> return sp.getFloat(key, (defaultObject as Float))
            "Long" -> return sp.getLong(key, (defaultObject as Long))
            else -> {
            }
        }
        return null
    }

    /**
     * 清除所有数据
     */
    fun clearAll() {
        val sp = CONTEXT.getSharedPreferences(FILE_NAME,
                Context.MODE_PRIVATE)
        val editor = sp.edit()
        editor.clear().apply()
    }

    /**
     * 清除指定数据
     */
    fun clear(key: String?) {
        val sp = CONTEXT.getSharedPreferences(FILE_NAME,
                Context.MODE_PRIVATE)
        val editor = sp.edit()
        editor.remove(key)
        editor.apply()
    }
}